### ISSUE TITLE
Can't install i2cssh with ruby 2.5.0 on OSX

### ISSUE THREAD

**Issue by joelsdc** [wouterdebie/i2cssh#83]
Installed ruby from brew, then tried installing i2cssh using gem:

```
$ gem install i2cssh
Building native extensions. This could take a while...
ERROR:  Error installing i2cssh:
	ERROR: Failed to build gem native extension.

    current directory: /usr/local/lib/ruby/gems/2.5.0/gems/rb-scpt-1.0.2
/usr/local/opt/ruby/bin/ruby -r ./siteconf20180328-57815-sjcbpk.rb extconf.rb
create /usr/local/lib/ruby/gems/2.5.0/gems/rb-scpt-1.0.2/src/osx_ruby.h ...
create /usr/local/lib/ruby/gems/2.5.0/gems/rb-scpt-1.0.2/src/osx_intern.h ...
creating Makefile

current directory: /usr/local/lib/ruby/gems/2.5.0/gems/rb-scpt-1.0.2
make "DESTDIR=" clean

current directory: /usr/local/lib/ruby/gems/2.5.0/gems/rb-scpt-1.0.2
make "DESTDIR="
compiling src/SendThreadSafe.c
compiling src/rbae.c
In file included from src/rbae.c:12:
In file included from src/osx_ruby.h:2546:
/usr/local/Cellar/ruby/2.5.0_2/include/ruby-2.5.0/ruby/backward.h:32:36: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]
DEPRECATED(void rb_autoload(VALUE, ID, const char*));
                                   ^
/usr/local/Cellar/ruby/2.5.0_2/include/ruby-2.5.0/ruby/backward.h:53:45: error: unknown type name 'ID'
DEPRECATED(int rb_frame_method_id_and_class(ID *idp, VALUE *klassp));
                                            ^
src/rbae.c:186:35: warning: implicit conversion loses integer precision: 'long' to 'AETransactionID' (aka 'int') [-Wshorten-64-to-32]
  AETransactionID transactionID = NUM2LONG(transactionIDValue);
                  ~~~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/osx_ruby.h:650:21: note: expanded from macro 'NUM2LONG'
#define NUM2LONG(x) RB_NUM2LONG(x)
                    ^~~~~~~~~~~~~~
src/osx_ruby.h:649:24: note: expanded from macro 'RB_NUM2LONG'
#define RB_NUM2LONG(x) rb_num2long_inline(x)
                       ^~~~~~~~~~~~~~~~~~~~~
src/rbae.c:493:9: warning: 'LSFindApplicationForInfo' is deprecated: first deprecated in macOS 10.10 - Use LSCopyApplicationURLsForBundleIdentifier instead. [-Wdeprecated-declarations]
  err = LSFindApplicationForInfo(inCreator,
        ^
/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LSInfoDeprecated.h:785:1: note: 'LSFindApplicationForInfo' has been explicitly marked deprecated here
LSFindApplicationForInfo(
^
src/rbae.c:501:9: warning: 'FSRefMakePath' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  err = FSRefMakePath(&outAppRef, path, PATH_MAX);
        ^
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4086:18: note: 'FSRefMakePath' has been explicitly marked deprecated here
extern OSStatus  FSRefMakePath(const FSRef *ref, UInt8 *path, UInt32 pathBufferSize)     __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
                 ^
src/rbae.c:514:9: warning: 'FSPathMakeRef' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  err = FSPathMakeRef((UInt8 *)StringValuePtr(path), &appRef, NULL);
        ^
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4115:18: note: 'FSPathMakeRef' has been explicitly marked deprecated here
extern OSStatus  FSPathMakeRef(const UInt8 *path, FSRef *ref, Boolean *isDirectory)        __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
                 ^
src/rbae.c:517:11: warning: 'GetNextProcess' is deprecated: first deprecated in macOS 10.9 [-Wdeprecated-declarations]
    err = GetNextProcess(&psn);
          ^
/System/Library/Frameworks/ApplicationServices.framework/Frameworks/HIServices.framework/Headers/Processes.h:452:1: note: 'GetNextProcess' has been explicitly marked deprecated here
GetNextProcess(ProcessSerialNumber * PSN)                     AVAILABLE_MAC_OS_X_VERSION_10_0_AND_LATER_BUT_DEPRECATED_IN_MAC_OS_X_VERSION_10_9;
^
src/rbae.c:519:11: warning: 'GetProcessBundleLocation' is deprecated: first deprecated in macOS 10.9 [-Wdeprecated-declarations]
    err = GetProcessBundleLocation(&psn, &foundRef);
          ^
/System/Library/Frameworks/ApplicationServices.framework/Frameworks/HIServices.framework/Headers/Processes.h:815:1: note: 'GetProcessBundleLocation' has been explicitly marked deprecated here
GetProcessBundleLocation(
^
src/rbae.c:521:9: warning: 'FSCompareFSRefs' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
    if (FSCompareFSRefs(&appRef, &foundRef) == noErr)
        ^
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:1850:15: note: 'FSCompareFSRefs' has been explicitly marked deprecated here
extern OSErr  FSCompareFSRefs(const FSRef *ref1, const FSRef *ref2)                     __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
              ^
src/rbae.c:533:9: warning: 'GetProcessForPID' is deprecated: first deprecated in macOS 10.9 [-Wdeprecated-declarations]
  err = GetProcessForPID(NUM2INT(pid), &psn);
        ^
/System/Library/Frameworks/ApplicationServices.framework/Frameworks/HIServices.framework/Headers/Processes.h:930:1: note: 'GetProcessForPID' has been explicitly marked deprecated here
GetProcessForPID(
^
src/rbae.c:546:9: warning: 'FSPathMakeRef' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  err = FSPathMakeRef((UInt8 *)StringValuePtr(path), &appRef, NULL);
        ^
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4115:18: note: 'FSPathMakeRef' has been explicitly marked deprecated here
extern OSStatus  FSPathMakeRef(const UInt8 *path, FSRef *ref, Boolean *isDirectory)        __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
                 ^
src/rbae.c:548:3: warning: 'LSApplicationParameters' is deprecated: first deprecated in macOS 10.10 - Use NSWorkspace instead. [-Wdeprecated-declarations]
  LSApplicationParameters appParams = {0,
  ^
/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LSOpenDeprecated.h:151:3: note: 'LSApplicationParameters' has been explicitly marked deprecated here
} LSApplicationParameters __OSX_AVAILABLE_BUT_DEPRECATED_MSG(__MAC_10_4, __MAC_10_10, __IPHONE_NA, __IPHONE_NA, "Use NSWorkspace instead.");
  ^
src/rbae.c:553:9: warning: 'LSOpenApplication' is deprecated: first deprecated in macOS 10.10 - Use -[NSWorkspace launchApplicationAtURL:options:configuration:error:] instead. [-Wdeprecated-declarations]
  err = LSOpenApplication(&appParams, &psn);
        ^
/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LSOpenDeprecated.h:183:1: note: 'LSOpenApplication' has been explicitly marked deprecated here
LSOpenApplication(
^
src/rbae.c:709:9: warning: 'FSPathMakeRef' is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
  err = FSPathMakeRef((UInt8 *)StringValuePtr(path), &fsRef, NULL);
        ^
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4115:18: note: 'FSPathMakeRef' has been explicitly marked deprecated here
extern OSStatus  FSPathMakeRef(const UInt8 *path, FSRef *ref, Boolean *isDirectory)        __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
                 ^
13 warnings and 1 error generated.
make: *** [rbae.o] Error 1

make failed, exit code 2

Gem files will remain installed in /usr/local/lib/ruby/gems/2.5.0/gems/rb-scpt-1.0.2 for inspection.
Results logged to /usr/local/lib/ruby/gems/2.5.0/extensions/x86_64-darwin-16/2.5.0/rb-scpt-1.0.2/gem_make.out
$
```

The versions brew installed:

```
$ gem -v
2.7.6
$ ruby -v
ruby 2.5.0p0 (2017-12-25 revision 61468) [x86_64-darwin16]
$
```

Any suggestions?

**Comment by joelsdc** [comment]
As a note, #81 complains about v2.4.3.

I can confirm i2cssh works on v2.4.2 (downgraded with `rvm` and was able to install & use i2cssh).

**Comment by jmoisesreyes** [comment]
Perhaps, try to uninstall old ruby from brew and then install less than or equal to version 2.4.2, The below example is with version 2.3
 
`#!/bin/bash
set -vex
 
brew uninstall ruby || true
brew uninstall ruby@2.3 || true
brew install ruby@2.3
 
sudo gem uninstall i2cssh
sudo gem install i2cssh
 
i2cssh`

**Comment by nynhex** [comment]
@joelsdc confirmed same issue on High Sierra with `2.5.0` try bumping to `2.5.1` it fixed my issue although the `2.5.0` problem still exists.  I can try to PR this if the group is ok with that.
